db/maintainer.js

```diff
@@ -13,23 +13,101 @@ class MaintainerDbController {
     return pkgs.map(({ _id: id, ...rest }) => ({ id, ...rest }))
   }
 
-  async getRevenue ({ maintainerId }) {
-    const packages = await this.db.collection('packages').find({
-      maintainers: { $elemMatch: { maintainerId } }
-    }).toArray()
-    return packages.reduce((totalRevenue, pkg) => {
-      const { revenuePercent } = pkg.maintainers.find((maintainer) => maintainer.maintainerId === maintainerId)
-      return totalRevenue + (pkg.totalRevenue * (revenuePercent / 100))
-    }, 0)
-  }
-
   async updateIlpPointer ({ userId, ilpPointer }) {
     return this.db.collection('users').updateOne({
       _id: ObjectId(userId)
     }, {
       $set: { 'payoutInfo.ilpPointer': ilpPointer }
     })
   }
+
+  // @returns an array of [{ _id: <packageId>, payout: <payoutInMS> }]
+  async getPendingPayout ({ maintainerId }) {
+    return this.db.collection('packages').aggregate([
+      {
+        $match: {
+          'maintainers.userId': {
+            $eq: maintainerId
+          }
+        }
+      }, {
+        $match: {
+          $or: [
+            {
+              adRevenue: {
+                $ne: null
+              }
+            }, {
+              donationRevenue: {
+                $ne: null
+              }
+            }
+          ]
+        }
+      }, {
```

Comment:
`$match` can accept multiple conditions; I think these two blocks can be combined.